src/app/Dashboard/DashboardLayoutToolbar.tsx

```diff
@@ -137,6 +142,34 @@ export const DashboardLayoutToolbar: React.FC<DashboardLayoutToolbarProps> = (_p
 
   const currLayout = React.useMemo(() => dashboardConfigs.layouts[dashboardConfigs.current], [dashboardConfigs]);
 
+  // first save
+  // Handle the click event of the "Clear All Cards" button
+  const handleClearAllCards = React.useCallback(() => {
+    // Dispatch an action to clear all cards for the selected layout
+    const updatedLayout = { ...currLayout, cards: [] };
+    dispatch(dashboardConfigCreateLayoutIntent(updatedLayout));
+  }, [dispatch, currLayout]);
+  
+  
+
+  const handleClearAllCardsWarningModalOpen = React.useCallback(
+    () => {
+      setSelectedLayoutForClear(currLayout.name);
+      setIsClearAllCardsConfirmationOpen(true);
+    },
+    [setSelectedLayoutForClear, setIsClearAllCardsConfirmationOpen, currLayout.name]
+  );
+  
+  const handleConfirmClearAllCards = React.useCallback(() => {
+    handleClearAllCards();
+    setIsClearAllCardsConfirmationOpen(false);
```

Comment:
For completeness, please also call `setSelectedLayoutForClear('');`

Same for `handleCancelClearAllCards`.